rskj-core/src/main/java/co/rsk/core/bc/BlockChainFlusher.java

```diff
@@ -67,6 +67,13 @@ public void start() {
     public void stop() {
         emitter.removeListener(listener);
         flushAll();
+        closeAll();
+    }
+
+    private void closeAll() {
+        logger.trace("closing blockStore.");
+        blockStore.close();
+        logger.trace("blockStore closed.");
```

Comment:
Should we add  `trieStore.dispose()`?

Reply:
Well... DataSourceWithCache does that job. I wouldn’t analyse that in the scope of this PR.

Reply:
Seems that `DataSource.close()` is not called on shutdown, and, as a result, `LevelDB` instances are not getting disposed gracefully.

In the examples [here](https://github.com/fusesource/leveldbjni) they do close db.

I guess we should investigate this more deeply in scope of this or another task.

Reply:
Yep, totally agreed. I will add another task to make a deep analysis.